monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,151 @@
+from monitoring.monitorlib.auth import make_auth_adapter
+from monitoring.monitorlib.infrastructure import DSSTestSession
+import asyncio
+from monitoring.monitorlib import rid
+import json, os
+import uuid
+from pathlib import Path
+from typing import  Any
+from monitoring.rid_qualifier.utils import OperatorLocation, RIDFlightDetails, TestFlightDetails, TestFlight, TestPayload, DeliverablePayload
+import arrow
+from uuid import UUID
+from typing import List
+from monitoring.rid_qualifier.utils import RIDQualifierTestConfiguration
+
+class TestBuilder():
+    ''' A class to setup the test data and create the objects ready to be submitted to the test harness '''
+
+    def __init__(self, test_configuration: RIDQualifierTestConfiguration) -> None:
+        self.test_configuration = test_configuration
+
+        self.aircraft_states_directory = Path('test_definitions', test_configuration.locale, 'aircraft_states')
+        
+        self.verify_aircraft_states_directory(self.aircraft_states_directory)
+        self.aircraft_state_files = os.listdir(self.aircraft_states_directory) 
+                
+        self.usses = self.test_configuration.usses
+        
+        self.disk_rid_state_data =[]
+        for uss_index, uss in enumerate(self.usses):
+            flight_track_path = Path(self.aircraft_states_directory, self.aircraft_state_files[uss['allocated_flight_track_number']])
+            
+            with open(flight_track_path) as generated_rid_state:
+                disk_rid_state_file = json.load(generated_rid_state)
+                self.disk_rid_state_data.append(disk_rid_state_file)
+        
+            
+    def verify_aircraft_states_directory(self, aircraft_states_directory) -> None:
+
+        ''' This method checks if there are tracks in the tracks directory '''
+        
+        files = [f for f in os.listdir(aircraft_states_directory) if os.path.isfile(os.path.join(aircraft_states_directory, f))]
+        if files:
+            pass
+        else:
+            raise ValueError("The there are no tracks in the tracks directory, create tracks first using the flight_data_generator module. ")
+
+    def make_json_compatible(self, struct: Any) -> Any:
+        if isinstance(struct, tuple) and hasattr(struct, '_asdict'):
+            return {k: self.make_json_compatible(v) for k, v in struct._asdict().items()}
+        elif isinstance(struct, dict):
+            return {k: self.make_json_compatible(v) for k, v in struct.items()}
+        elif isinstance(struct, str):
+            return struct
+        try:
+            return [self.make_json_compatible(v) for v in struct]
+        except TypeError:
+            return struct
+
+    def build_test_payload(self) ->List[DeliverablePayload]: 
```

Comment:
nit: Probably `build_test_payloads` (plural) since it builds multiple payloads

Reply:
Thanks now this has been renamed.